### PR TITLE
Add type definitions for split-file

### PR DESCRIPTION
**Pull request by dolanmiu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Repo link:
https://github.com/tomvlk/node-split-file

NPM link:
https://www.npmjs.com/package/split-file

### CODE REVIEW

File: types/split-file/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
```
Comment (peterblazejewicz) [inline types/split-file/tsconfig.json:6]:
```suggestion
            "es6"
```
you've @types/ndoe as dependency, why `dom` lib has been added?

Comment (dolanmiu) [reply types/split-file/tsconfig.json:6]: Because `console.log` would give errors otherwise

Comment (dolanmiu) [reply types/split-file/tsconfig.json]: I will try commiting the suggestion

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dolanmiu thanks!

**Comment by dolanmiu** [comment]
Ready to merge

**Comment by dolanmiu** [comment]
Ready to merge

**Pull request merged**